src/Compositor.cpp

```diff
@@ -2150,7 +2153,7 @@ void CCompositor::moveWorkspaceToMonitor(CWorkspace* pWorkspace, CMonitor* pMoni
 
     // fix old mon
     int nextWorkspaceOnMonitorID = -1;
-    if (!SWITCHINGISACTIVE || !POLDMON)
+    if (!SWITCHINGISACTIVE)
```

Comment:
why is this check removed

Reply:
`const bool SWITCHINGISACTIVE = POLDMON ? POLDMON->activeWorkspace == pWorkspace->m_iID : false;`

`POLDMON` can never be `false` when `SWITCHINGISACTIVE` is `true` due to the definition of `SWITCHINGISACTIVE`. So the check is redundant.

Should I open a separate PR for this change ?

Reply:
right my bad